crates/compiler/src/ast.rs

```diff
@@ -0,0 +1,260 @@
+        Self {
+            expr_type: ExprType::Identifier,
+            token: None,
+            ident: Some(ident),
+            left: None,
+            right: None,
+            paren_args: vec![],
+        }
+    }
+
+    pub fn new_terminal(expr_type: ExprType, token: Token) -> Self {
+        Self {
+            expr_type: expr_type,
+            token: Some(token),
+            ident: None,
+            left: None,
+            right: None,
+            paren_args: vec![],
+        }
+    }
+
+    pub fn new_unary(expr_type: ExprType, child: Expr) -> Self {
+        Self {
+            expr_type: expr_type,
+            token: None,
+            ident: None,
+            left: Some(Box::new(child)),
+            right: None,
+            paren_args: vec![],
+        }
+    }
+
+    pub fn new_binary(expr_type: ExprType, left: Expr, right: Expr) -> Self {
+        Self {
+            expr_type: expr_type,
+            token: None,
+            ident: None,
+            left: Some(Box::new(left)),
+            right: Some(Box::new(right)),
+            paren_args: vec![],
+        }
+    }
+
+    pub fn new_function_call(name: Identifier, paren_args: Vec<Expr>) -> Self {
+        Self {
+            expr_type: ExprType::FunctionCall,
+            token: None,
+            ident: Some(name),
+            left: None,
+            right: None,
+            paren_args,
+        }
+    }
+
+    pub fn new_tuple_or_paren(args: Vec<Expr>) -> Self {
+        Self {
+            expr_type: ExprType::TupleOrParen,
+            token: None,
+            ident: None,
+            left: None,
+            right: None,
+            paren_args: args,
+        }
+    }
+}
+
+impl Debug for Expr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.fmt_with_indent(f, 0)
+    }
+}
+
+impl Expr {
+    fn fmt_with_indent(&self, f: &mut fmt::Formatter<'_>, indent: usize) -> fmt::Result {
+        write!(f, "{:indent$}", "", indent = indent * 2)?;
+
+        // Print token if present
+        if let Some(token) = &self.token {
+            write!(f, " '{}'", token.lexeme)?;
+        }
+
+        // Print identifier if present
+        if let Some(ident) = &self.ident {
+            write!(f, " '{}'", ident.token.lexeme)?;
+        }
+
+        let has_children =
+            self.left.is_some() || self.right.is_some() || !self.paren_args.is_empty();
+        if has_children {
+            writeln!(f)?;
+
+            // Print left child if present
+            if let Some(left) = &self.left {
+                left.fmt_with_indent(f, indent + 1)?;
+                writeln!(f)?;
+            }
+
+            // Print right child if present
+            if let Some(right) = &self.right {
+                right.fmt_with_indent(f, indent + 1)?;
+                writeln!(f)?;
+            }
+
+            // Print args if present
+            for (_i, arg) in self.paren_args.iter().enumerate() {
```

Comment:
If the iteration count is not used, no need to use `enumerate()`

```suggestion
            for arg in self.paren_args.iter() {
```